File: app/src/main/java/com/assignment/data/apis/ApiServices.java
package com.assignment.data.apis;

import com.assignment.data.SearchPhotoDataModel;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiServices {

    @GET("/services/rest/")
    Single<SearchPhotoDataModel> searchPhotos(@Query("method") String method
            , @Query("api_key") String apiKey
            , @Query("text") String text
            , @Query("per_page") int pageSize
            , @Query("page") int page
            , @Query("format") String format
            , @Query("nojsoncallback") int noJsonCallback);

}

File: app/src/main/java/com/assignment/framework/apis/ApiServices.kt
package com.apis

import com.model.SearchPhotoDataModel
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiServices {

    @GET("/services/rest/")
    fun searchPhotos(@Query("method") method: String, @Query("api_key") apiKey: String,
                     @Query("text") text: String, @Query("per_page") pageSize: Int,
                     @Query("page") page: Int, @Query("format") format: String,
                     @Query("nojsoncallback") noJsonCallback: Int): Single<SearchPhotoDataModel>

}
